example-app/src/main/java/com/adyen/checkout/example/repositories/paymentMethods/PaymentsRepository.kt

```diff
@@ -15,25 +15,27 @@ import com.adyen.checkout.example.data.api.model.CancelOrderRequest
 import com.adyen.checkout.example.data.api.model.CreateOrderRequest
 import com.adyen.checkout.example.data.api.model.PaymentMethodsRequest
 import com.adyen.checkout.example.data.api.model.PaymentsRequest
+import com.adyen.checkout.example.data.api.model.PaymentsRequestData
 import com.adyen.checkout.example.data.api.model.SessionRequest
 import com.adyen.checkout.example.repositories.safeApiCall
 import com.adyen.checkout.sessions.model.Session
-import okhttp3.ResponseBody
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import org.json.JSONObject
-import retrofit2.Call
 
 interface PaymentsRepository {
     suspend fun getSessionAsync(sessionRequest: SessionRequest): Session?
     suspend fun getPaymentMethods(paymentMethodsRequest: PaymentMethodsRequest): PaymentMethodsApiResponse?
-    fun paymentsRequest(paymentsRequest: PaymentsRequest): Call<ResponseBody>
-    suspend fun paymentsRequestAsync(paymentsRequest: PaymentsRequest): ResponseBody?
-    fun detailsRequest(detailsRequest: JSONObject): Call<ResponseBody>
-    suspend fun detailsRequestAsync(detailsRequest: JSONObject): ResponseBody?
-    suspend fun balanceRequestAsync(request: BalanceRequest): ResponseBody?
-    suspend fun createOrderAsync(orderRequest: CreateOrderRequest): ResponseBody?
-    suspend fun cancelOrderAsync(request: CancelOrderRequest): ResponseBody?
+    fun paymentsRequest(paymentsRequest: PaymentsRequest): JSONObject?
+    suspend fun paymentsRequestAsync(paymentsRequest: PaymentsRequest): JSONObject?
+    fun detailsRequest(detailsRequest: JSONObject): JSONObject?
+    suspend fun detailsRequestAsync(detailsRequest: JSONObject): JSONObject?
+    suspend fun balanceRequestAsync(request: BalanceRequest): JSONObject?
+    suspend fun createOrderAsync(orderRequest: CreateOrderRequest): JSONObject?
+    suspend fun cancelOrderAsync(request: CancelOrderRequest): JSONObject?
 }
 
+@Suppress("TooManyFunctions")
```

Comment:
This lint check annoys me the most 😆  I keep thinking of adding a global rule to ignore it. Do you have any ideas why we could benefit from keeping it?

Reply:
Most of the times this check is a good indicator that your class is doing too many different things. For this repository for example it could be split up in session, payment methods, payment request, details, balance and order repositories. For Android classes (Fragment, Activity and maybe ViewModel) this check doesn't really make sense, but I believe it's possible to disable it for those